src/content/resources/cpp/multidimensional-arrays.md

```diff
@@ -0,0 +1,122 @@
+
+Example 3: 
+```c++
+int arr[2][3] = {{1,2,3}, {4,5,6}};
+```
+This will make a 2x3 matrix that looks like: 
+```c++
+{{1,2,3},{4,5,6}}
+```
+
+##### Note:
+It is possible to go ever further, and so a declaration like `data_type array_name[A][B][C]` is possible and would add more layers to your array. So after going through one layer, you have to input the location of the next layer, and so on and so forth until you reach the deepest layer of the `{}` where you'll find the object. 
+
+Example 4: 
+```c++
+int arr[2][3][4];
+``` 
+This will make an 2x3x4 matrix that looks like: 
+```c++
+{{{0,0,0,0},{0,0,0,0},{0,0,0,0}},{{0,0,0,0},{0,0,0,0},{0,0,0,0}}}
+```
+
+### What can you do with it?
+
+##### Calling objects
+To get the object inside, you can call for the object inside a multidimensional array by putting a integer inside the `[]`.
+
+So taking the array made in example 2 above: 
+```c++
+int arr[2][3] = {{1,2,3}, {4,5,6}};
+std::cout << arr[0][1]; 
+```
+This would output `2` because using array indicies, `2` is the object in position 1 of the 0th array.
+
+You must put an integer in both `[]`, or else you will get an error. 
+There is no syntax to instantly view the whole array, but an easy way is to loop through the whole array and outputting it.  
+
+##### Reassigning objects
+Like with a regular array, you can also reassign what object is being held inside a exact position using `=`
+So let's take for example the array made in example 2 above, which looks like: 
+```c++
+int arr[2][3] = {{1,2,3}, {4,5,6}};
+arr[1][1] = 1337;
+```
+This would cause the array to change to looking like:
+```c++
+{{1,2,3},{4,1337,6}}
+```
+
+### Restrictions
+
+1. The size is locked upon initialization. Once you make the array, its size is locked. For example, if you make a 2x3 array, and realize you need another column or row, you have to either make a new one, or change the size where you initialize it.
+2. There is no way to search except by iterating through each object in the entire array. For example, if you need to find the number `2` in `{{1,2,3},{4,5,6}}`, you need to go through every object in a nested `for` loop to find the positions where `2` is located.
+3. No way to organize. Unlike with a `class`, you cannot designate a column with something like a name or title. For example, if you want to store student IDs and their grade, you have to note that column 1 is the student ID and column 2 is their grade.
+4. No on-demand sorting. Unlike other containers, multidimensional arrays do not come with a built-in sorting function, such as highest to lowest or vice versa. It is stuck in that format unless you create your own sorting functions that take the array and sort it.
```

Comment:
Maybe mention that the standard library data structures like `std::vector` do have built in sorting functions

Reply:
Okay, I added that in.